### DISCUSSION TITLE
[V5] whereHas is not a function

### DISCUSSION THREAD

**Discussion by hariprasath-yadav** [adonisjs/core#1899] [Help] [answered]
I have tables for a many-to-many relationship
branches, users, branch_user

likewise, I created the model for branches and users
and i have set the below-given function in Branch.ts Model
```
  @manyToMany(() => User)
  public users: ManyToMany<typeof User>
```
if i try to get all the branches which have users as an engineer like
```
const engBranches = await Branches.query().whereHas('users', (qry) => {
  qry.where('designation', 'engineer')
})
```
it shows an error 

> Branch_1.default.query(...).whereHas is not a function

**Comment by thetutlage** [comment]
Can you share the complete model and the version of `adonisjs/lucid` package?

**Reply by hariprasath-yadav** [reply]
OK but for easy understand, i changed some code in question
but my original code will differ like originally i used testUser not user
I attached the model as files
and my original query is
```
serEmp.testUser.id = 1 // it is dynamic 
await Branch.query().whereHas('testUsers', (qry) => {
      qry.where('id', serEmp.testUser.id)
    })
```
and verion
> "@adonisjs/lucid": "^8.2.2"

and i uploaded model files
[Branch.ts](https://github.com/adonisjs/core/files/5531645/Branch.txt)
[TestUser.ts](https://github.com/adonisjs/core/files/5531646/TestUser.txt)

**Reply by hariprasath-yadav** [reply]
I created a new discussion for this problem because it does not end with this one function
and the discussion link is
[Querying relationship existence is not working](https://github.com/adonisjs/core/discussions/1901)

**Comment by hariprasath-yadav** [comment, marked as answer]
Solution in
[Querying relationship existence is not working](https://github.com/adonisjs/core/discussions/1901)
